core-plugins/src/test/java/co/cask/hydrator/plugin/batch/action/OracleExportActionTest.java

```diff
@@ -0,0 +1,170 @@
+
+/**
+ * Test for {@link OracleExportAction}
+ */
+public class OracleExportActionTest extends ETLBatchTestBase {
+  // Test keypair config
+  @ClassRule
+  public static TemporaryFolder folder = new TemporaryFolder();
+  private static final String PASSPHRASE = "password";
+  private static final String[] SHELL_ARGS = {"/bin/sh", "-i", "-l"};
+  private static final String USER = "test";
+  // Test members
+  private SshServer sshServer;
+  private static MiniDFSCluster dfsCluster;
+  private static FileSystem fileSystem;
+
+  @BeforeClass
+  public static void buildMiniDFS() throws Exception {
+    // Setup MiniDFSCluster
+    File baseDir = folder.newFolder();
+    Configuration conf = new Configuration();
+    conf.set(MiniDFSCluster.HDFS_MINIDFS_BASEDIR, baseDir.getAbsolutePath());
+    MiniDFSCluster.Builder builder = new MiniDFSCluster.Builder(conf);
+    dfsCluster = builder.build();
+    dfsCluster.waitActive();
+    fileSystem = FileSystem.newInstance(conf);
+  }
+
+  @AfterClass
+  public static void cleanupMiniDFS() throws Exception {
+    // Shutdown MiniDFSCluster
+    dfsCluster.shutdown();
+    folder.delete();
+  }
+
+  @Before
+  public void setup() throws Exception {
+    sshServer = setupSSHDaemon();
+    sshServer.start();
+  }
+
+  @After
+  public void cleanup() throws Exception {
+    if (sshServer != null) {
+      sshServer.stop();
+    }
+  }
+
+  private SshServer setupSSHDaemon() throws Exception {
+    SshServer sshServer = SshServer.setUpDefaultServer();
+    SimpleGeneratorHostKeyProvider simpleGeneratorHostKeyProvider = new SimpleGeneratorHostKeyProvider();
+    simpleGeneratorHostKeyProvider.setAlgorithm(KeyUtils.RSA_ALGORITHM);
+    sshServer.setKeyPairProvider(simpleGeneratorHostKeyProvider);
+    sshServer.setPasswordAuthenticator(new PasswordAuthenticator() {
+      @Override
+      public boolean authenticate(String username, String password, ServerSession session) {
+        return username.equals(USER) && password.equals(PASSPHRASE);
+      }
+    });
+    sshServer.setShellFactory(new ProcessShellFactory(SHELL_ARGS));
+    sshServer.setCommandFactory(EchoCommandFactory.INSTANCE);
+    return sshServer;
+  }
+
+  @Test
+  public void testCheckWrongArgumentForServerPassword() throws Exception {
+    OracleExportAction.OracleExportActionConfig oracleExportActionConfig =
+      new OracleExportAction.OracleExportActionConfig(sshServer.getHost(),
+                                                      sshServer.getPort(), USER, "wrongPassword", "system",
+                                                      "cask", "home", "sid", "/tmp/junit.csv",
+                                                      "select * from test;", "csv");
+    OracleExportAction oracleExportAction = new OracleExportAction(oracleExportActionConfig);
+    MockActionContext mockActionContext = new MockActionContext();
+    try {
+      oracleExportAction.run(mockActionContext);
+    } catch (IOException e) {
```

Comment:
Can we throw a more specific exception instead so we don't have to rely on verifying the error message.